src/package-hoister.js

```diff
@@ -107,19 +117,12 @@ export default class PackageHoister {
         return sortAlpha(aPattern, bPattern);
       });
 
-      //
-      const infos = [];
       for (const [pattern, parent] of queue) {
         const info = this._seed(pattern, {isDirectRequire: false, parent});
         if (info) {
-          infos.push(info);
+          this.hoist(info);
         }
       }
-
-      //
-      for (const info of infos) {
-        this.hoist(info);
-      }
```

Comment:
fyi, this was unrelated, (so is the for loop below) but the second iteration really didn't seem necessary. lmk if they're need for some reason.